harness/determined/common/experimental/experiment.py

```diff
@@ -166,6 +168,34 @@ def delete(self) -> None:
         """
         bindings.delete_DeleteExperiment(self._session, experimentId=self._id)
 
+    def download_code(self, output_dir: Optional[str] = None) -> str:
+        """Downloads a zipped tarball (*.tar.gz) of the experiment's submitted code locally.
+
+        If a file with the same name already exists in the output directory, overwrites the file.
+
+        Arguments:
+            output_dir (string, optional): The local directory path to save downloaded archive to,
+                creating directory if it does not exist. If unspecified, will save to current
+                working directory.
+
+        Returns:
+             Filepath of downloaded code archive.
+        """
+        resp = bindings.get_GetModelDef(self._session, experimentId=self._id)
+        output_filename = f"exp-{self.id}_model_def.tar.gz"
```

Comment:
FWIW, `det e download-model-def` downloads a file called `experiment-{id}_model_def.tgz`.

Reply:
yeah, i didn't like that name for two reasons:
1.  it's actually `experiment_{id}_model_def.tgz` with underscores between "experiment" and "{id}" but i feel it's more readable for "experiment id" to be separate from "model_def" since the former is its own "word".
2. though `.tgz` and `.tar.gz` are equivalent, i think the latter is more expressive of a file extension (the file is a `tar` zipped by `gzip`). 

it does bother me that this would be inconsistent with the CLI. however, the CLI is already inconsistent with the Web UI, which names this file `exp32_mnist_pytorch_const_model_def.tar.gz`. i found that name to be overly verbose. i dunno, let me know if you have preferences and/or suggestions for a different/better name altogether.

Reply:
Exact same comment here as above.